libs/sevenseg/sevenseg.ts

```diff
@@ -0,0 +1,435 @@
+enum SegmentStyle {
+    //% block="blank"
+    Blank = 0,
+    //% block="thin"
+    Thin = 1,
+    //% block="narrow"
+    Narrow = 2,
+    //% block="medium"
+    Medium = 3,
+    //% block="thick"
+    Thick = 4
+}
+
+enum SegmentScale {
+    //% block="full"
+    Full,
+    //% block="half"
+    Half
+}
+/**
+* Seven segment display digits and gizmos
+*/
+//% icon="\uf2a1" color="#4682B4" blockGap=8
+//% groups='["Create", "Counter", "Digits", "Position"]'
+namespace sevenseg {
+
+    const fullSegment: number[][] = [
+        [1, 0, 14, 0, 2, 1, 13, 1, 3, 2, 12, 2, 4, 3, 11, 3],
```

Comment:
Do these arrays change or are they just a lookup table? If they are the latter, you can save some memory by converting them into hex buffers.

Here's a one-liner to convert an array:

```javascript
"hex`" + myArray.map(n => n.toString(16)).map(n => n.length < 2 ? "0" + n : n).join("") + "`;"
```

Then you can just replace the array with the output of that expression in your code

Reply:
For example:

```typescript
const fullSegment: Buffer[] = [
    hex`abcde`,
    hex`abcdef`,
]
```

Reply:
They are ordinate metrics for the segments so they stay fixed. I'll play with your suggestion. Thanks!